packages/manager/modules/pci/src/projects/project/storages/databases/database/users/add/add.html

```diff
@@ -20,32 +19,65 @@ <h2 data-translate="pci_databases_users_add"></h2>
     />
 </oui-field>
 
-<h2 data-translate="pci_databases_users_add_roles"></h2>
-<oui-dual-list
-    data-source="$ctrl.availableRoles"
-    data-target="$ctrl.model.selectedRoles"
-    property="name"
->
-    <oui-dual-list-source
-        data-heading="{{:: 'pci_databases_users_add_roles_available' | translate }}"
-        data-searchable="true"
-    >
-    </oui-dual-list-source>
-    <oui-dual-list-target
-        data-heading="{{:: 'pci_databases_users_add_roles_selected' | translate }}"
-        data-searchable="true"
+<div ng-if="!$ctrl.isRolesReadOnly">
+    <h2 data-translate="pci_databases_users_add_roles"></h2>
+    <oui-dual-list
+        data-source="$ctrl.availableRoles"
+        data-target="$ctrl.model.selectedRoles"
+        disabled="$ctrl.isFeatureActivated('getRoles')"
+        property="name"
     >
-    </oui-dual-list-target>
-</oui-dual-list>
+        <oui-dual-list-source
+            data-heading="{{:: 'pci_databases_users_add_roles_available' | translate }}"
+            data-searchable="true"
+        >
+        </oui-dual-list-source>
+        <oui-dual-list-target
+            data-heading="{{:: 'pci_databases_users_add_roles_selected' | translate }}"
+            data-searchable="true"
+        >
+        </oui-dual-list-target>
+    </oui-dual-list>
+</div>
+
+<div ng-if="$ctrl.database.engine === 'redis'">
```

Comment:
This kind of code should be moved to a modal like in this case it might be Database.class. This will help us to scale and add additional databases easily.

Reply:
Done here: https://github.com/ovh/manager/pull/5488/commits/45d16da89ff2c70503a9a111c3fd5996a5eecbe1